pybamm/quick_plot.py

```diff
@@ -11,182 +29,181 @@ class QuickPlot(object):
+            raise NotImplementedError("cannot plot 3D variables")
 
     def reset_axis(self):
         """
         Reset the axis limits to the default values.
         """
-        self.axis = {
-            "Negative particle surface concentration": [0, self.l_n, 0, 1],
-            "Electrolyte concentration": [
-                0,
-                1,
-                np.min(self.c_e(self.t, self.x) - 0.2),
-                np.max(self.c_e(self.t, self.x) + 0.2),
-            ],
-            "Positive particle surface concentration": [1 - self.l_p, 1, 0, 1],
-            "Total current density": [
-                self.t[0],
-                self.t[-1],
-                np.min(self.i_cell(self.t)) - 1,
-                np.max(self.i_cell(self.t)) + 1,
-            ],
-            "Negative electrode potential [V]": [
-                0,
-                self.l_n,
-                np.min(self.phi_s_n(self.t, self.x_n)) - 0.01,
-                np.max(self.phi_s_n(self.t, self.x_n)) + 0.01,
-            ],
-            "Electrolyte potential [V]": [
-                0,
-                1,
-                np.min(self.phi_e(self.t, self.x)),
-                np.max(self.phi_e(self.t, self.x)),
-            ],
-            "Positive electrode potential [V]": [
-                1 - self.l_p,
-                1,
-                np.min(self.phi_s_p(self.t, self.x_p)),
-                np.max(self.phi_s_p(self.t, self.x_p)),
-            ],
-            "Terminal voltage [V]": [
-                self.t[0],
-                self.t[-1],
-                np.min(self.V(self.t)),
-                np.max(self.V(self.t)),
-            ],
-        }
+        self.axis = {}
+        for name, variable in self.variables.items():
+            if variable[0].dimensions == 1:
+                y_min = np.min([ax_min(v(self.ts[i])) for i, v in enumerate(variable)])
+                y_max = np.max([ax_max(v(self.ts[i])) for i, v in enumerate(variable)])
+                if y_min == y_max:
+                    y_min -= 1
+                    y_max += 1
+                self.axis[name] = [self.min_t, self.max_t, y_min, y_max]
+            elif variable[0].dimensions == 2:
+                x = self.x_values[name]
+                x_scaled = x * self.x_scale
+                y_min = np.min(
+                    [ax_min(v(self.ts[i], x)) for i, v in enumerate(variable)]
+                )
+                y_max = np.max(
+                    [ax_max(v(self.ts[i], x)) for i, v in enumerate(variable)]
+                )
+                if y_min == y_max:
+                    y_min -= 1
+                    y_max += 1
+                self.axis[name] = [x_scaled[0], x_scaled[-1], y_min, y_max]
 
     def plot(self, t):
         """Produces a quick plot with the internal states at time t.
 
         Parameters
         ----------
         t : float
-            Time at which to plot.
+            Dimensional time at which to plot.
```

Comment:
If you implement my previous comment, then we should also state that this must be in hours aswell. Unless we want to pass an optional flag to QuickPlot and set `self.time_dimensions`